src/components/BlogContextMenu.js

```diff
@@ -0,0 +1,74 @@
+import React from 'react';
+
+import { Dropdown, DropdownToggle, DropdownMenu, DropdownItem, Collapse } from 'reactstrap';
+
+class BlogContextMenu extends React.Component {
+  constructor(props) {
+    super(props);
+    this.toggle = this.toggle.bind(this);      
+    this.state = {
+      dropdownOpen: false,
+      collapseOpen: false
+    };
+  }  
+  
+  toggle(event) {
+    this.setState({
+      dropdownOpen: !this.state.dropdownOpen
+    });
+    updateBlog();
+  }
+  
+  clickEdit(post) {    
+    csrfFetch("/cube/blogsrc/" + post._id)
+      .then(response => response.json())
+      .then(function(json) {
+        if (json.src) {
+          $('#editor').html(json.src);
+        } else {
+          $('#editor').html(json.body);
+        }
+
+        $('#postBlogTitleInput').val(json.title);
```

Comment:
Optimally this state would be controlled by react - we would have an element `<div contenteditable value={...} onChange={...} />`. But this is fine for now.